### ISSUE TITLE
Update c-scrollbar_thumb height on resize

### ISSUE THREAD

**Issue by jmartinezuk** [locomotivemtl/locomotive-scroll#17]
Hi, awesome plugin. One of the issues I've found is that the thumb height of '.c-scrollbar_thumb' is not updated on window resizes. It happens on the test site as well.
I don't know if it's related, but the positions of the elements are not updated as well when resizing the window, only after you do a manual scroll. I tried calling scroll.update(), on a window resize listener, but it didn't work. 

How do I get everything updated after resizing the window?
Thanks!

**Comment by tiansial** [comment]
I have a similar issue, where I need to resize the window to get all the elements in the viewport.
https://streamable.com/0dlgf

**Comment by jmartinezuk** [comment]
I've found that if you force a scroll to on load to the top of the container, it will force the code to run and show you the content. I was running to the very same thing as well. I solved it like this:

`let contentWrapper = document.querySelector('.content-wrapper')
    const scroll = new locomotiveScroll({
        el: contentWrapper,
        smooth: true,
        smoothMobile: true
    })
    scroll.scrollTo('.scroll-content',0) // .scroll-content is my first child inside .content-wrapper`

**Comment by tiansial** [comment]
Not working for me.

**Comment by GojkoC** [comment]
I have the same problem. scroll won't go to the bottom of the page. It happens only after I resize the browser.

**Comment by tiansial** [comment]
> I have the same problem. scroll won't go to the bottom of the page. It happens only after I resize the browser.

Are you using it with browserify?

**Comment by GojkoC** [comment]
No, I managed to fix it but it is a hack. It only works when I set fixed height to the scrolling area and it works, but it doesn't feel right to me.

**Comment by tiansial** [comment]
> No, I managed to fix it but it is a hack. It only works when I set fixed height to the scrolling area and it works, but it doesn't feel right to me. I'm using it in react, with webpack.

Do you mind sharing a snippet? 

In a related note, I'm getting trouble in updating the scrolling area if I expand a accordion-like element.

**Comment by jmartinezuk** [comment]
I've found that firing a resize event several times on actions that expand or contract the page, sorts most of the issues regarding sizing out.

**Comment by tiansial** [comment]
> I've found that firing a resize event several times on actions that expand or contract the page, sorts most of the issues regarding sizing out.

window.dispatchEvent(new Event('resize')); ?

**Comment by jmartinezuk** [comment]
Yeah, put in a function that fires it say 20 times every 0.2 seconds and you should be good to go. It's a hack, but it works.

**Comment by tiansial** [comment]
In my case, I was using GSAP and I was able to fix the resize problem with onUpdate callback

```
toggleTl = new TimelineLite({
        onUpdate: () => {
          window.dispatchEvent(new Event('resize'));
        },
        paused: true,
      }),
```

**Comment by jmartinezuk** [comment]
As long as it doesnt run continuously, you're all good. I believe just firing it on the onComplete should work in your case.

**Comment by adamcoulombe** [comment]
There is an internal `update` function that is called on resize. I have not tested, but it would seem we should be calling `this.reinitScrollBar()` [here](https://github.com/locomotivemtl/locomotive-scroll/blob/29d07a8af14e4d9f81212a79e0ec208dbe93d451/src/scripts/Smooth.js#L540). (Similar to how `this.initsScrollBar` is [called within the init function](https://github.com/locomotivemtl/locomotive-scroll/blob/29d07a8af14e4d9f81212a79e0ec208dbe93d451/src/scripts/Smooth.js#L69)).

I suspect this was the intent because as far as I can tell, the function  `this.reinitScrollBar` exists but is not actually being used anywhere.

**Comment by adamcoulombe** [comment]
per my previous comment, it looks like there is already a PR from @Julien-prrs that should resolve this:
https://github.com/locomotivemtl/locomotive-scroll/pull/35

**Comment by 2frags** [comment]
> I have a similar issue, where I need to resize the window to get all the elements in the viewport.
> https://streamable.com/0dlgf

I have exactly the same issue. And PR from @Julien-prrs not didn`t helped for me.
For now i use little hack with window resize & setTimeout:

`function lscrollrefresh(func) {
    $(window).trigger('resize');
    func.update();
}
setTimeout(lscrollrefresh, 200, lscroll); `

And thanks author for the great plugin :)

**Comment by Julien-prrs** [comment]
For a recent project, I used that code to deal with the problem.
`
const _scroll = new LocomotiveScroll({
    el: el,
    smooth: true,
    smoothMobile: false,
    inertia: 1.1
});

function update() {
    _scroll.update();
    _scroll.scroll.reinitScrollBar();
}
`

This code worked well for me

**Comment by adamcoulombe** [comment]
> > I have a similar issue, where I need to resize the window to get all the elements in the viewport.
> > https://streamable.com/0dlgf
> 
> I have exactly the same issue. And PR from @Julien-prrs didn`t helped for me.
> For now i use little hack with window resize & setTimeout:
> 
> `function lscrollrefresh(func) { $(window).trigger('resize'); func.update(); } setTimeout(lscrollrefresh, 200, lscroll); `
> 
> And thanks author for the great plugin :)

@2frags the issue described in the original post is different than your issue. The original post is about the scrollbar thumb position. You may want to consider opening a seperate PR for the problem you have.

**Comment by quentinhocde** [comment]
Hi!

The main issue is fixed in the version `3.2.2` in this commit : f98c0d8 

Thanks :)
